sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala

```diff
@@ -422,8 +422,7 @@ class TungstenAggregationIterator(
         metrics.incPeakExecutionMemory(maxMemory)
 
         // Update average hashmap probe if this is the last record.
-        val averageProbes = hashMap.getAverageProbesPerLookup()
-        avgHashmapProbe.add(averageProbes.ceil.toLong)
+        avgHashProbe.set(hashMap.getAverageProbesPerLookup())
```

Comment:
The probe for non-whole-stage HashAggregate is updated here. Actually, I now think we should move this update to the end of processing input rows, instead of at the last record. Because even the iterator is not consumed to the end (e.g., a Limit after Aggregate), we still should show the metric.